roles/ec2_instance_create/README.md

```diff
@@ -0,0 +1,157 @@
+# ec2_instance_create
+
+A role to create an EC2 instance in AWS.
+
+Users can specify various parameters for instance configuration, including instance type, AMI ID, key pair, tags, and VPC/subnet configuration.
+
+This role also supports the creation of optional networking resources, such as an external security group and an Elastic IP (EIP). You can choose to wait for the EC2 instance to finish booting before continuing.
+
+## Role Variables
+
+The following variables can be set in the role to customize EC2 instance creation and networking configurations:
+
+### Role Operation
+
+* **ec2_instance_create_operation**: (Optional)
+  Whether to create or delete resources using the role. Default is `create`.
+  Choices are `create` and `delete`.
+
+### EC2 Instance Configuration
+
+* **ec2_instance_create_aws_region**: (Required)
+  The AWS region in which to create the EC2 instance.
+
+* **ec2_instance_create_instance_name**: (Required)
+  The name of the EC2 instance to be created.
+
+* **ec2_instance_create_instance_type**: (Required)
+  The instance type for the EC2 instance (e.g., `t2.micro`, `m5.large`).
+
+* **ec2_instance_create_ami_id**: (Required)
+  The AMI ID for the EC2 instance.
+
+* **ec2_instance_create_key_name**: (Required)
+  The name of the key pair to use for SSH access to the EC2 instance.
+  If the key does not exist, a key pair will be created with the name.
+
+* **ec2_instance_create_vpc_subnet_id**: (Required)
+  The ID of the VPC subnet in which the instance will be launched.
+
+* **ec2_instance_create_tags**: (Optional)
+  A dictionary of tags to assign to the EC2 instance.
+
+* **ec2_instance_create_wait_for_boot**: (Optional)
+  Whether to wait for the EC2 instance to be in the "running" or "terminated" state before continuing. Default is `true`.
+
+### Optional Networking Resources
+
+#### Elastic IP
+
+* **ec2_instance_create_vpc_id**: (Optional)
```

Comment:
I think it would useful having it around for additional flexibility to users. But fine with removing if everyone in agreement. 
cc @alinabuzachis @GomathiselviS

Reply:
Ha, I just started to add a similar comment as Aubin, we should just use the VPC of the ec2 instance.